src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/volume/v2/OpenStackVolumePluginTest.java

```diff
@@ -1,201 +1,524 @@
+        this.plugin.doRequestInstance(endpoint, jsonRequest, cloudUser);
 
         // verify
-        Mockito.verify(this.client).doDeleteRequest(Mockito.anyString(), Mockito.any(OpenStackV3User.class));
+        Mockito.verify(this.client, Mockito.times(TestUtils.RUN_ONCE)).doPostRequest(Mockito.eq(endpoint),
+                Mockito.eq(jsonRequest), Mockito.eq(cloudUser));
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).doGetVolumeResponseFrom(jsonResponse);
     }
-
-    // test case: Deleting an instance without a project ID must raise FogbowException.
-    @Test(expected = FogbowException.class)
-    public void testRemoveInstanceWithoutProjectId() throws Exception {
+    
+    // test case: When calling the doGetVolumeResponseFrom method with a JSON
+    // malformed, it must verify that a UnexpectedException was throw.
+    @Test
+    public void testDoGetVolumeResponseFromJsonMalformed() {
+        // set up
+        String json = JSON_MALFORMED;
+        String expected = Messages.Error.ERROR_WHILE_GETTING_VOLUME_INSTANCE;
+
+        try {
+            // exercise
+            this.plugin.doGetVolumeResponseFrom(json);
+            Assert.fail();
+        } catch (UnexpectedException e) {
+            // verify
+            Assert.assertEquals(expected, e.getMessage());
+        }
+    }
+    
+    // test case: When calling the doGetResponseFromCloud method and an unexpected error
+    // occurs, it must verify that the map method of the
+    // OpenStackHttpToFogbowExceptionMapper class has been called.
+    @Test
+    public void testDoGetResponseFromCloudFail() throws Exception {
         // set up
-        this.openStackV3Token.setProjectId(null);
+        OpenStackV3User cloudUser = createOpenStackUser();
+        String endpoint = generateEndpoint(cloudUser.getProjectId(), OpenStackVolumePlugin.VOLUMES,
+                TestUtils.FAKE_INSTANCE_ID);
+
+        HttpResponseException expectedException = new HttpResponseException(ERROR_STATUS_CODE, MESSAGE_STATUS_CODE);
+        Mockito.doThrow(expectedException).when(this.client).doGetRequest(Mockito.eq(endpoint), Mockito.eq(cloudUser));
+
+        PowerMockito.mockStatic(OpenStackHttpToFogbowExceptionMapper.class);
+        PowerMockito.doCallRealMethod().when(OpenStackHttpToFogbowExceptionMapper.class, MAP_METHOD, Mockito.any());
+        
+        try {
+            // exercise
+            this.plugin.doGetResponseFromCloud(endpoint, cloudUser);
+            Assert.fail();
+        } catch (Exception e) {
+            // verify
+            PowerMockito.verifyStatic(OpenStackHttpToFogbowExceptionMapper.class, Mockito.times(TestUtils.RUN_ONCE));
+            OpenStackHttpToFogbowExceptionMapper.map(Mockito.eq(expectedException));
+        }
+    }
+    
+    // test case: When calling the findVolumeTypeId method, it must verify
+    // that the call was successful.
+    @Test
+    public void testFindVolumeTypeId() throws FogbowException {
```

Comment:
actually this method has to be better tested. Verify its calls is not enough. You should also verify the returned value. That's because the method do some computations against the parameters besides the calls.

Reply:
Understood ... I'll try to improve this test.